eduvpn/ui/ui.py

```diff
@@ -447,36 +459,37 @@ def enter_ConnectionStatus(self, old_state, new_state):
         self.update_connection_server()
         self.update_connection_status()
 
-        if hasattr(self, '_cancel_expiry_updates'):
+        if hasattr(self, '_cancel_validity_updates'):
             # Cancel any previous threads, as they might have been cancelled
             # and there shouln't be multiple threads running.
-            self._cancel_expiry_updates()
+            self._cancel_validity_updates()
 
-        def update_connection_expiry():
+        def update_connection_validity():
             # This function runs in a background thread.
             state = self.app.interface_state
             if not isinstance(state, interface_state.ConnectionStatus):
                 # cancel this thread
                 return False
-            run_in_main_gtk_thread(self.update_connection_expiry)()
-            if datetime.utcnow() < state.expiry:
+            run_in_main_gtk_thread(self.update_connection_validity)()
+            if datetime.utcnow() < state.validity.end:
                 return True
             else:
                 self.app.network_transition_threadsafe('set_certificate_expired')
                 return False
 
-        self._cancel_expiry_updates = run_periodically(
-            update_connection_expiry,
+        self._cancel_validity_updates = run_periodically(
+            update_connection_validity,
             UPDATE_EXIPRY_INTERVAL,
-            'update-expiry',
+            'update-validity',
         )
 
     @transition_edge_callback(EXIT, interface_state.ConnectionStatus)
     def exit_ConnectionStatus(self, old_state, new_state):
         self.show_component('connectionPage', False)
 
-        self._cancel_expiry_updates()
-        del self._cancel_expiry_updates
+        if hasattr(self, '_cancel_validity_updates'):
```

Comment:
this looks a bit like a hack, isn't there another way to accomplish this? With a boolean flag? I just find a bit odd the change the signature of an object runtime.

Reply:
Yes, this bit is not pretty. I will create a proper mechanism for this later.